presto-hive/src/test/java/com/facebook/presto/hive/TestMergeJoinPlan.java

```diff
@@ -336,14 +340,16 @@ private PlanMatchPattern joinPlan(String leftTableName, String rightTableName, L
                         joinType,
                         joinClauses.build(),
                         Optional.empty(),
-                        PlanMatchPattern.tableScan(leftTableName, leftColumnReferencesBuilder.build()),
-                        PlanMatchPattern.tableScan(rightTableName, rightColumnReferencesBuilder.build()))) :
+                        exchange(LOCAL, GATHER,
+                                tableScan(leftTableName, leftColumnReferencesBuilder.build())),
+                        exchange(LOCAL, GATHER,
+                                tableScan(rightTableName, rightColumnReferencesBuilder.build())))) :
```

Comment:
Can you help me understand why local exchange could work in such senario?
In my understanding, without grouped execution, we need remote exchange to reshuffle the data between workers

Reply:
All the data for a given join key will be colocated on the same node (that's true for any kind of join), so we should just need a local exchange to merge all the already sorted input. 

The thing that might not work about this (have to run an end-to-end test still), is if you can have one task reading from multiple splits (that come from different buckets).  Then the data streaming into this exchange from each task won't be already sorted. We really need to have this merging exchange applied for each split in order to avoid adding an extra sort.  I'm not sure what's needed to accomplish that.